apps/src/blockly/addons/cdoSpritePointer.js

```diff
@@ -0,0 +1,112 @@
+// Helper functions for pointer block logic. In our pointer blocks (a.k.a.)
+// mini toolbox blocks) we "shadow" the image on the block from a source block.
+// This file contains functions to get and set the image on the pointer block.
+
+/**
+ * Get the image url for the given pointer block.
+ * @param {Block} block pointer block to get the image url for
+ * @param {*} pointerMetadataMap Object of the form {blockType: {imageSourceType: <string>, imageIndex: <number>}},
+ *  which maps pointer block types to the image source block type they can shadow and the index of the image to shadow.
+ * @param {string} imageSourceId Optional id of the image source block.
+ * @returns The url of the image that the pointer block should display, or an empty string if the block should display no image
+ */
+export function getPointerBlockImageUrl(
+  block,
+  pointerMetadataMap,
+  imageSourceId
+) {
+  const pointerData = pointerMetadataMap[block.type];
+  if (!pointerData || !block.inputList || block.inputList.length === 0) {
+    return '';
+  }
+  let imageSource = undefined;
```

Comment:
It's not immediately apparent to me that this will reference a block. Could we add a comment for rename to `imageSourceBlock`?

Reply:
done!